jetbrains-core/src/software/aws/toolkits/jetbrains/core/toolwindow/ToolkitToolWindow.kt

```diff
@@ -4,135 +4,118 @@
 package software.aws.toolkits.jetbrains.core.toolwindow
 
 import com.intellij.openapi.Disposable
-import com.intellij.openapi.components.ServiceManager
+import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.util.Disposer
+import com.intellij.openapi.util.Key
+import com.intellij.openapi.wm.RegisterToolWindowTask
 import com.intellij.openapi.wm.ToolWindow
-import com.intellij.openapi.wm.ToolWindowAnchor
 import com.intellij.openapi.wm.ToolWindowManager
 import com.intellij.ui.content.Content
-import com.intellij.ui.content.impl.ContentImpl
-import icons.AwsIcons
-import java.util.concurrent.ConcurrentHashMap
-import javax.swing.Icon
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.withContext
+import software.aws.toolkits.jetbrains.utils.getCoroutineUiContext
 import javax.swing.JComponent
 
 interface ToolkitToolWindow {
```

Comment:
I think we can go further and kill this more.

Let's follow the SQS tool window pattern and wrap CW and CFN in project services, and have those manage their own tool window, registered through the tool window EP.